### ISSUE TITLE
Where are the ECMAScript communities and enthusiasts?

### ISSUE THREAD

**Issue by kytrinyx** [exercism/javascript#409]
As we move towards the launch of the new version of Exercism we are going to be ramping up on actively recruiting people to help provide feedback.

Our goal is to get to 100%: everyone who submits a solution and wants feedback should get feedback. Good feedback. You can read more about this aspect of the new site here: http://mentoring.exercism.io/

To do this, we're going to need a lot more information about where we can find language enthusiasts.

* Is ECMAScript supported by one or more large organizations?
* Does ECMAScript have an official community manager?
* Do you know of specific communities (online or offline) that are enthusiastic about ECMAScript? (Chat communities, forums, meetups, student clubs, etc)
* Are there popular conferences for ECMAScript? (If so, what are some examples?)
* Are there any organizations who are targeted specifically at getting certain subgroups or demographics interested in ECMAScript? (e.g. kids, teenagers, career changers, people belonging to various groups that are typically underrepresented in tech?)
* Are there specific groups or programs dedicated to mentoring people in ECMAScript?
* Are there popular newsletters for ECMAScript?
* Is ECMAScript taught at programming bootcamps? (If so, what are some examples?)
* Is ECMAScript taught at universities? (If so, what are some examples?)

In other words: where do people care a lot and/or know a lot about ECMAScript?

This is part of the project being tracked in https://github.com/exercism/meta/issues/103

**Comment by SleeplessByte** [comment]
In general, online or offline, people will refer to ECMAScript as JavaScript and generally _not know_ the difference between the language specification (ECMAScript) and the various implementations. ECMAScript in general is not taught, or "chatted" about, but in reality it's what we mean most of the time when we say "...well in JS you can X, Y, Z".

> Is ECMAScript supported by one or more large organizations?

- Under the [linux foundation projects](https://www.linuxfoundation.org/projects/), there is the [JS foundation](https://js.foundation/)
- ECMAScript (originally Sun Microsystems) is now _designed_ and developed by the [TC39 Royalty Free Task Group](https://www.ecma-international.org/memento/tc39-rf-tg.htm
), Ecma International, Technical Committee 39, with members from _many_ big organisations vested, such as Microsoft, Google, Paypal and so forth.

It has various implementation which are all supported by a large or multiple large companies. Here is a list of various well and lesser known implementations. The [list of engines](https://en.wikipedia.org/wiki/List_of_ECMAScript_engines) is much longer, but this should suffice for now.
- [ActionScript](https://www.adobe.com/devnet/actionscript.html) by Adobe (*note*: reaching [EOL in 2020](https://blogs.adobe.com/conversations/2017/07/adobe-flash-update.html))
- [ChakraCore](https://github.com/Microsoft/ChakraCore) by Microsoft (*note*: [EOL](https://blogs.windows.com/windowsexperience/2018/12/06/microsoft-edge-making-the-web-better-through-more-open-source-collaboration/#IIycUFupVTBcbAY5.97) when Chakra is replaced by Chromium)
- [Chrome V8](https://v8.dev/) by Google
- [JScript](https://docs.microsoft.com/en-us/previous-versions//hbxc2t98(v=vs.85)) by Microsoft (*note*: reached EOL with IE 10)
- [Nashorn](https://blogs.oracle.com/nashorn/) by Oracle
- [SpiderMonkey](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey) by Mozilla
- [SquirrelFish](https://trac.webkit.org/wiki/SquirrelFish) by Apple (*note*: now part of WebKit)
- [QTScript](https://doc.qt.io/archives/qt-4.8/scripting.html) by Nokia (*note*: reached EOL, taken over by the QT Project, and dropped passed QT 4.8) 
- [Webkit's JavaScriptCore / Nitro](https://webkit.org/) by Apple, Adobe, Google, KDE

Because several languages, frameworks, engines and combinations are built on top of the ECMAScript specification (or an implementation -> e.g. Node on V8), these are also supporting the underlying ECMAScript. A few are:
- [TypeScript](https://en.wikipedia.org/wiki/TypeScript) by Microsoft
- [NodeJS](https://nodejs.org/en/) by the Node.js Foundation (Joylent, and Linux Foundation)

> Does ECMAScript have an official community manager?

I don't know what this question really entails. The thing that comes closest is the TC39 group as listed above.

> Do you know of specific communities (online or offline) that are enthusiastic about ECMAScript? (Chat communities, forums, meetups, student clubs, etc).
> Are there popular conferences for ECMAScript? (If so, what are some examples?)

I would say that apart from the TC39 task group, and the [RFCs and contributions](https://github.com/tc39/ecma402) around it, [talks and panels](https://www.youtube.com/watch?v=Hj5q8uyqGYc), there are various groups and communities that talk about specific implementations. There are various conferences and followings:

- [BrazilJS](https://braziljs.org/conf/)
- [dotJS](https://2019.dotjs.io/)
- [EmpireJS](https://2018.empirejs.org/)
- [Holy JS](https://holyjs-moscow.ru/en/)
- [International JavaScript Conference](https://javascript-conference.com/)
- [JSConf](https://jsconf.com/) in all its formats
- [JS Fest](http://jsfest.com.ua/)
- [JS Heroes](https://jsheroes.io/)
- [JS Kongress](https://js-kongress.com/)
- [JS Nation](https://amsterdamjs.com/)
- [JS Poland](https://js-poland.pl/)
- [Midwest JS](http://www.midwestjs.com/#/main)
- [Nordic.js](https://nordicjs.com/2019)
- [Odessa.js](http://odessajs.org/)
- [Ruhr.js](https://ruhrjs.de/)
- [Scotland.js](http://scotlandjs.com/)

This does _not_ include all the specific communities revolving around the fast ecosystems, frameworks and platforms built with and on top of JavaScript (and therefore ultimately ECMAScript). Think about Vue, React (native), Express and Node specific communities (such as [nodeconf](http://nodeconf.com/) and [event by the linux foundation](https://events.linuxfoundation.org/events/node-js-interactive-2018/), but also [Framework Summit](https://www.frameworksummit.com/) and [connect.tech](http://connect.tech/) and conferences such as [perf.now](https://perfnow.nl/)).

Several shows cover JavaScript or specific implementation updates, such as Chrome Developers' [HTTP 203](https://www.youtube.com/playlist?list=PLNYkxOF6rcIAKIQFsNbV0JDws_G_bnNo9). Additionally, there are several resources to learn the language and there are discussions in the mostly OSS exercise repos (think [javascript.info](https://javascript.info/) and [learn-js.org](https://www.learn-js.org/)).

I'm specifically listing _conferences_ and not websites because these places many communities come together as one. There are _so many_ facebook groups and so forth which are comparatively in scale not very important. Personally I don't know of any _large_ communities discussing just ECMAScript.

> Are there any organizations who are targeted specifically at getting certain subgroups or demographics interested in ECMAScript? (e.g. kids, teenagers, career changers, people belonging to various groups that are typically underrepresented in tech?)

I think you're looking for something like [railsgirls](http://railsgirls.com/).

- [Girl Develop IT](https://www.girldevelopit.com/materials/intro-js) has material on JavaScript
- [JS Girls](http://js-girls.org/) was supposed to be a one day workshop for women but hasn't had activity in the past 2 years as far as I can tell. [Girls.js](https://girlsjs.pl/) however _did_ pickup.
- There are regular meetup groups such as [Girls Who JavaScript in bangalore](https://www.meetup.com/Girls-Who-Javascript/), and the ones [listed here](https://medium.com/@sitapati/women-in-javascript-ec4035fe2290).
- I believe that there is now a bootcamp in various places in Africa, but I don't know the names.
- The [Career Change Coders](https://medium.com/career-change-coder/) write about JavaScript quite a bit.
- There a several books and platforms for JavaScript for kids, but I wouldn't take them seriously and as far as I can tell they're all not actively developed for.

> Are there specific groups or programs dedicated to mentoring people in ECMAScript?

- Apart from this one, there are a lot of _individuals_ doing code mentoring in ECMAScript, sites offer tutorials (also guided / mentored) such as [tree house](https://teamtreehouse.com/library/topic:javascript), and the general code mentor sites such as [codementor.io](
https://www.codementor.io/). I haven't found any that are _specifically_ javascript only.
- [Codewars](https://www.codewars.com/kata/javascript) has a _lot_ on JavaScript
- [CoderDojo](https://coderdojo.com/) has them on JavaScript as well

> Are there popular newsletters for ECMAScript?

- [JavaScript weekly](https://javascriptweekly.com/)

There are also newsletters not _specific_ to ECMAScript, but will include them such as:
- [Frontend Focus](https://frontendfoc.us/)
- [Smashing Magazine](https://www.smashingmagazine.com/the-smashing-newsletter/)
- [Think with Google](https://www.thinkwithgoogle.com/)

> Is ECMAScript taught at programming bootcamps? (If so, what are some examples?)

I'm just going to list 15 random of them, ordered by name, but basically _almost all of them_:
- [Actualize](https://anyonecanlearntocode.com/chicago-coding-bootcamp)
- [App Academy](https://www.appacademy.io/immersive/curriculum/weeks/6-9?location=san-francisco#2)
- [Bloc](https://www.bloc.io/web-developer-track)
- [Coding Dojo](https://www.codingdojo.com/onsite-boot-camp) (only basic JavaScript)
- [Epicodus](https://www.epicodus.com/what-to-expect)
- [Flatiron School](https://flatironschool.com/our-courses/)
- [General Assembly](https://generalassemb.ly/education/javascript-development)
- [Hack Reactor](https://www.hackreactor.com/onsite-immersive)
- [Hacker You](https://hackeryou.com/courses/bootcamp)
- [Iron Hack](http://www.ironhack.com/en/courses/web-development-bootcamp)
- [Lambda School](https://lambdaschool.com/courses/cs/web/)
- [Le Wagon](https://www.lewagon.com/program)
- [Tech Talent South](https://www.techtalentsouth.com/courses/javascript-101/)
- [The Tech Academy](https://learncodinganywhere.com/codingbootcamps)
- [Thinkful](https://www.thinkful.com/bootcamp/web-development/full-time/)

> Is ECMAScript taught at universities? (If so, what are some examples?)

What constitutes as a University is _not_ the same world wide, but in general in the Netherlands, BSc tracks for Computer Science will teach JavaScript as part of their non-elective curriculum.

**Comment by aprilmintacpineda** [comment]
> Is ECMAScript supported by one or more large organizations?

Does http://nodejs.org count? I think this question is a bit hard to answer since AFAIK ECMAScript is a trademark created, by Ecma International, to standardize JavaScript. So is it supported by a large organization? Absolutely YES!

> Does ECMAScript have an official community manager?

Hmmm... ECMAScript is NOT a community.

> Do you know of specific communities (online or offline) that are enthusiastic about ECMAScript? (Chat communities, forums, meetups, student clubs, etc)

Here in PH, I know a few one

- https://www.meetup.com/ManilaJavaScript/
- https://www.meetup.com/reactjs-developers-manila/
- https://www.meetup.com/D3-jsPH/
- https://www.meetup.com/Women-Who-Code-Manila/
- https://www.meetup.com/ngmanila/
- https://www.meetup.com/Pasig-City-node-js-Meetup/
- https://www.meetup.com/Meteor-Manila/

> Are there popular conferences for ECMAScript? (If so, what are some examples?)

The groups cited about conduct FREE and PAID events and conferences on a monthly routine.

> Are there any organizations who are targeted specifically at getting certain subgroups or demographics interested in ECMAScript? (e.g. kids, teenagers, career changers, people belonging to various groups that are typically underrepresented in tech?)

Those who are interested.

> Are there specific groups or programs dedicated to mentoring people in ECMAScript?

Here in the Philippines, none that I know of. Though there are PAID courses that people can enrol to.

> Are there popular newsletters for ECMAScript?

https://javascriptweekly.com/

> Is ECMAScript taught at programming bootcamps? (If so, what are some examples?)

Here in the Philippines, none that I know of. Though there are PAID courses that people can enrol to.

> Is ECMAScript taught at universities? (If so, what are some examples?)

AFAIK, no.

**Comment by kytrinyx** [comment]
@SleeplessByte This is epic, thank you.
@aprilmintacpineda This is really useful additional context, much appreciated!

**Comment by junedev** [comment]
I am closing this issue, we will still refer back to the provided information though.
